src/Nuvem.PharmacyManagementSystem.Pharmacy.Services/PharmacyService.cs

```diff
@@ -13,34 +14,42 @@ public interface IPharmacyService
 public class PharmacyService : IPharmacyService
 {
     private readonly IPharmacyRepository _pharmacyRepository;
-    private readonly IMapper _mapper;
 
-    public PharmacyService(IPharmacyRepository pharmacyRepository, IMapper mapper)
+    public PharmacyService(IPharmacyRepository pharmacyRepository)
     {
-        _pharmacyRepository = pharmacyRepository;
-        _mapper = mapper;
+        _pharmacyRepository = pharmacyRepository;        
     }
 
     public async Task<List<PharmacyModel>> GetAllAsync()
     {     
-        List<Data.EFEntities.Pharmacy> pharmacyList = await _pharmacyRepository.GetAllAsync();
-        return _mapper.Map<List<PharmacyModel>>(pharmacyList);
+        List<PharmacyEntity> pharmacyList = await _pharmacyRepository.GetAllPharmaciesAsync();
+        
+        if(pharmacyList is null) return null;
+        List<PharmacyModel> resultList = new();
+        foreach(var pharmacy in pharmacyList)
+        {
+            var ph = MappingHelper<PharmacyModel,PharmacyEntity>.Map(pharmacy);
```

Comment:
Just curious, what is the benefit of having both a PharmacyModel and a PharmacyEntity?

Reply:
> Just curious, what is the benefit of having both a PharmacyModel and a PharmacyEntity?

I would also ask is it necessary to say PharmacyEntity (instead of just keeping as Pharmacy)? Generally when it comes from the EF - i like to keep it as-is namewise to avoid confusion...

Reply:
Also consider if those models should be partial class extension of the EF-provided entity.
It would then mean that your models and EF Entities would need to share the same namespace - which is typically allowed.

Reply:
And finally - your Model - in many cases - you dont really need a model to express every little variation of the read-only responses that your API sends back to the UI. I recommend leveraging anonymous objects for the 1-off custom models that the UI may enjoy consuming... just things to consider

Reply:
> Just curious, what is the benefit of having both a PharmacyModel and a PharmacyEntity?

PharmacyModel is DTO(data transfer object) used by service layer to get the data from data layer and transfer to api layer. PharmacyEntity is the Pharmacy Entity created by Entity framework. I have to use the alias(PharmacyEntity) as Pharmacy type is clashing with Pharmacy namespace in the code.

Reply:
> > Just curious, what is the benefit of having both a PharmacyModel and a PharmacyEntity?
> 
> I would also ask is it necessary to say PharmacyEntity (instead of just keeping as Pharmacy)? Generally when it comes from the EF - i like to keep it as-is namewise to avoid confusion...

Agree. I had to use the alias PharmacyEntity to avoid the clash between the Pharmacy type and Pharmacy namespace.
Lesson Learned: Avoid type name as part of namespace.